src/datachain/lib/utils.py

```diff
@@ -28,3 +30,29 @@ def __init__(self, message):
 class DataChainColumnError(DataChainParamsError):
     def __init__(self, col_name, msg):
         super().__init__(f"Error for column {col_name}: {msg}")
+
+
+def normalize_col_names(col_names: Sequence[str]) -> dict[str, str]:
+    gen_col_counter = 0
+    new_col_names = {}
+    org_col_names = set(col_names)
+
+    for org_column in col_names:
+        new_column = org_column.lower()
+        new_column = re.sub("[-_\\s]+", "_", new_column)
+        new_column = new_column.strip("_")
+        new_column = re.sub("[^0-9a-z_]+", "", new_column)
```

Comment:
```suggestion
        new_column = org_column.lower()
        new_column = re.sub("[^0-9a-z]+", "_", new_column)
        new_column = new_column.strip("_")
```

Wouldn't you prefer this alternative normalization, which interprets any non-alphanumeric character as a separator?
* Before: `Old/Man's@Haven 1` → `oldmanshaven_1`
* After: `Old/Man's@Haven@ 1` → `old_man_s_haven_1`

Reply:
I'm fixing this (and a bunch of other case) - don't spend time yet on this please :)

Reply:
not sure, I'll think a bit about this ... there could be non ascii literals - should we also change them to _ - not sure tbh.

Reply:
You're right; we risk going too far with normalization, although I wonder if it'd be a good idea to create table names containing non-ASCII characters anyway.

We can always normalize diacritics with something like this:

```python
>>> from unicodedata import normalize
>>> normalize("NFKD", "próspera piraña").encode("ascii", "ignore")
"prospera pirana"
```

But, again, thorough normalization is a complex topic, and we'd still not be taking into account e.g. Cyrillic, Han...

Reply:
Irrelevant rambling (continued)...

```suggestion
        new_column = org_column.casefold()  # Unicode-friendly .lower()
        new_column = regex.sub("[^\p{L}\p{Nd}]+", "_", new_column)  # Unicode-friendly [^a-z0-9]
        new_column = new_column.strip("_")
```

Requires the `regex` package to use Unicode character classes and crossing your fingers for database support of non-ASCII column names.

Reply:
Based on the latest logic, I still believe my original proposal — https://github.com/iterative/datachain/pull/542#discussion_r1817963778 — would be an improvement over the current approach.

Reply:
Based on the latest logic, I still believe my original proposal — https://github.com/iterative/datachain/pull/542#discussion_r1817963778 — would be an improvement over the current approach.

Reply:
changed to replace with `_` all "noisy" symbols, let's not go into things like unicode normalization, etc 

is there anything left here?